redundant-bmc/src/sibling_impl.cpp

```diff
@@ -0,0 +1,293 @@
+        co_return object.begin()->first;
+    }
+    catch (const sdbusplus::exception_t&)
+    {}
+
+    co_return std::string{};
+}
+
+// namespace util
+
+void SiblingImpl::loadFromPropertyMap(
+    const SiblingImpl::PropertyMap& propertyMap)
+{
+    auto it = propertyMap.find("BMCPosition");
+    if (it != propertyMap.end())
+    {
+        bmcPosition = std::get<size_t>(it->second);
+    }
+
+    it = propertyMap.find("FWVersion");
+    if (it != propertyMap.end())
+    {
+        fwVersion = std::get<std::string>(it->second);
+    }
+
+    it = propertyMap.find("Provisioned");
+    if (it != propertyMap.end())
+    {
+        provisioned = std::get<bool>(it->second);
+    }
+
+    it = propertyMap.find("RedundancyEnabled");
+    if (it != propertyMap.end())
+    {
+        redEnabled = std::get<bool>(it->second);
+    }
+
+    it = propertyMap.find("FailoversPaused");
+    if (it != propertyMap.end())
+    {
+        failoversPaused = std::get<bool>(it->second);
+    }
+
+    it = propertyMap.find("BMCState");
+    if (it != propertyMap.end())
+    {
+        bmcState = std::get<BMCState>(it->second);
+    }
+
+    it = propertyMap.find("Role");
+    if (it != propertyMap.end())
+    {
+        role = std::get<Role>(it->second);
+    }
+
+    it = propertyMap.find("CommunicationOK");
+    if (it != propertyMap.end())
+    {
+        commsOK = std::get<bool>(it->second);
+    }
+
+    it = propertyMap.find("Heartbeat");
+    if (it != propertyMap.end())
+    {
+        heartbeat = std::get<bool>(it->second);
+    }
+}
+
+// NOLINTNEXTLINE
+sdbusplus::async::task<> SiblingImpl::watchInterfaceAdded()
+{
+    namespace rules = sdbusplus::bus::match::rules;
+    sdbusplus::async::match match(ctx,
+                                  rules::interfacesAddedAtPath(objectPath));
+
+    while (!ctx.stop_requested())
+    {
+        auto [_, interfaces] =
+            co_await match
+                .next<sdbusplus::message::object_path, InterfaceMap>();
+
+        auto it = interfaces.find(redundancy_ns::Sibling::interface);
+        if (it != interfaces.end())
+        {
+            lg2::info("Sibling D-Bus interface added");
+            loadFromPropertyMap(it->second);
+            interfacePresent = true;
+
+            // If first time seen, wait for the service name to get into
+            // the mapper and then start the nameOwnerChanged watch.
+            if (serviceName.empty())
+            {
+                size_t count = 0;
+                do
+                {
+                    using namespace std::chrono_literals;
+                    co_await sdbusplus::async::sleep_for(ctx, 100ms);
+                    serviceName = co_await getServiceName();
+                    lg2::info(
+                        "After interfacesAdded, sibling service is {SERVICE}",
+                        "SERVICE", serviceName);
+                } while (serviceName.empty() && (count++ < 200));
```

Comment:
Always been a fan of more descriptive constants for things like this (200).

Reply:
done.